Addition of Markdown preview to the markdown type text editor element

ijeh-i opened the pull request:
<!--
  Thanks for submitting a pull request!
  We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request.

  Before submitting a pull request, please make sure the following is done:
  1. Fork [the repository](https://github.com/finos/legend-studio) and create your branch from `master`.
  2. Run `yarn` in the repository root.
  3. If you've fixed a bug or added code that should be tested, add tests!
  4. Add a changeset to summarize your change in the changelogs.
  5. If you haven't already, complete the CLA.

  Learn more about contributing: https://github.com/finos/legend-studio/blob/master/CONTRIBUTING.md
-->

## Summary

Fixes https://github.com/finos/legend-studio/pull/1282

<!--
  Explain the **motivation** for making this change. What existing problem(s) does this PR solve?
  Also, link the issue(s) to this PR: e.g. Fixes #1, Resolves #2
  If you also added documentation, link the PR from https://github.com/finos/legend
-->

## How did you test this change?

- [ N ] Test(s) added
- [ Y ] Manual testing (please provide screenshots/recordings)
- [ Y ] No testing (please provide an explanation)

<!--
  Demonstrate the code is solid. Screenshots/videos if the pull request changes the user interface.
  How exactly did you verify that your PR solves the issue you wanted to solve?
  If you leave this empty, your PR will very likely be closed.
-->

<!--
  More in-depth docs below:
  - Git workflow: https://github.com/finos/legend-studio/blob/master/docs/workflow/working-with-github.md
  - Test: https://github.com/finos/legend-studio/blob/master/docs/technical/test-strategy.md
  - Changeset: https://github.com/finos/legend-studio/blob/master/



CONTRIBUTING.md#changeset
  - Dependency: https://github.com/finos/legend-studio/blob/master/docs/workflow/dependencies.md
  - UX/UI: https://github.com/finos/legend-studio/tree/master/docs/ux
-->


https://user-images.githubusercontent.com/52004757/178755894-a38a52d9-4977-46e7-b793-602065749b09.mov

MauricioUyaguari commented:
Also can you add the video in the PR summary itself. Additionally lets add the issue this PR would close.

ijeh-i commented:
@MauricioUyaguari  I have updated the Screen record on the summary.  Please review again

akphi commented:
Looking great @ijeh-i ! I think we should use a _2-state_ button here, basically something similar to the `hackermode` button, you can click on it to show the preview panel and click again to hide the panel away

ijeh-i commented:
@akphi  Thanks.. I believe i am currently using a 2-State button.. It’s the same single button that toggles the show and hide of the preview tab

akphi commented:
@ijeh-i I meant in terms of styling, can we do this? 

https://github.com/finos/legend-studio/blob/4746ab4188f782e858a9ad3484d9d53128ddab63/CONTRIBUTING.md

ijeh-i commented:
@akphi  ok .. I’ll update it to be a toggle icon button.  
Should i still leave the button where it is .. or should i place it at the bottom blue bar .. 
Currently i like where i placed it, seeing as it’s only effective for the text markup type element

ijeh-i commented:
@akphi  @MauricioUyaguari  I believe i have made the UI changes and also added the persistence of the preview panel if the tabs are changed. 
I have also updated the Video on the summary. 
Please review again.

akphi commented on .changeset/thirty-wolves-heal.md:
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-extension-dsl-text': patch
```
```suggestion
'@finos/legend-extension-dsl-text': minor
```

akphi commented on .changeset/thirty-wolves-heal.md:
```diff
@@ -0,0 +1,5 @@
+'@finos/legend-extension-dsl-text': patch
+---
+
+Addition of preview feature for markdown type text editor element.
```
```suggestion
Support previewing `Markdown` in editor
```

ijeh-i replied:
Done

akphi commented on packages/legend-extension-dsl-text/src/components/studio/TextElementEditor.tsx:
```diff
@@ -55,13 +62,22 @@ export const TextElementEditor = observer(() => {
   const textElement = textEditorState.textElement;
   const isReadOnly = textEditorState.isReadOnly;
   const typeNameRef = useRef<HTMLInputElement>(null);
+  const [showPreview, setShowPreview] = useState(
```
Or another thought is we could have a Preview state for the editor, and its plugin-based, if the type is something we have support for (for example, `markdown`, we show the preview support), else, we disable it. I'm fine with either approach

ijeh-i replied:
Ok .. I have created too separate components `renderMarkdownView` & `renderPlainTextEditorView`. 
So depending on the `isMarkdown`, it will display one of the views.

akphi commented on packages/legend-extension-dsl-text/src/models/metamodels/pure/model/packageableElements/text/DSLText_Text.ts:
```diff
@@ -24,13 +24,15 @@ import {
 export class Text extends PackageableElement implements Hashable {
   type?: string | undefined;
   content!: string;
+  preview?: boolean | undefined;
```
This is also the reason why you're failing the tests

ijeh-i replied:
@MauricioUyaguari  I placed it in the text metamodel so I can persist the toggle state of the preview mode. 

But I have reverted to using the state component of the TextElement to toggle the markdown preview mode.
Thank you.

ijeh-i commented on packages/legend-extension-dsl-text/src/stores/studio/DSLText_GraphModifierHelper.ts:
```diff
@@ -24,3 +24,7 @@ export const text_setType = action((text: Text, type: string): void => {
   text.content = content;
 });
+
+export const text_setPreview = action((text: Text, preview: boolean): void => {
```
Done

ijeh-i commented on .changeset/proud-mugs-turn.md:
```diff
@@ -0,0 +1,5 @@
+'@finos/legend-art': patch
+---
+
+Addition of vsc open preview icon.
```
Done
